drivers/bus/softqspi.c

```diff
@@ -166,11 +166,12 @@ STATIC void mp_soft_qspi_write_cmd_data(void *self_in, uint8_t cmd, size_t len,
     CS_HIGH(self);
 }
 
-STATIC void mp_soft_qspi_write_cmd_addr_data(void *self_in, uint8_t cmd, uint32_t addr, size_t len, const uint8_t *src) {
+STATIC void mp_soft_qspi_write_cmd_addr_data(void *self_in, uint8_t cmd, uint32_t addr, uint8_t addr_bytes, size_t len, const uint8_t *src) {
     mp_soft_qspi_obj_t *self = (mp_soft_qspi_obj_t*)self_in;
-    uint8_t cmd_buf[4] = {cmd, addr >> 16, addr >> 8, addr};
+    uint32_t addr_buff = __REV(addr);
     CS_LOW(self);
-    mp_soft_qspi_transfer(self, 4, cmd_buf, NULL);
+    mp_soft_qspi_transfer(self, 1, &cmd, NULL);
+    mp_soft_qspi_transfer(self, addr_bytes, (uint8_t*)&addr_buff, NULL);
```

Comment:
I would prefer to combine cmd and addr bytes to minimise the number of calls to `mp_soft_qspi_transfer`, to improve speed.